include/cadmium/celldevs/coupled/cells_coupled.hpp

```diff
@@ -0,0 +1,133 @@
+/**
+* Copyright (c) 2020, Román Cárdenas Rodríguez
+* ARSLab - Carleton University
+* GreenLSI - Polytechnic University of Madrid
+* All rights reserved.
+*
+* Standard coupled Cell-DEVS model
+*/
+
+#ifndef CADMIUM_CELLDEVS_CELLS_COUPLED_HPP
+#define CADMIUM_CELLDEVS_CELLS_COUPLED_HPP
+
+#include <exception>
+#include <string>
+#include <utility>
+#include <vector>
+#include <unordered_map>
+#include <cadmium/modeling/ports.hpp>
+#include <cadmium/modeling/dynamic_model.hpp>
+#include <cadmium/modeling/dynamic_model_translator.hpp>
+#include <cadmium/celldevs/cell/cell.hpp>
+
+
+namespace cadmium::celldevs {
+    /**
+     * standard coupled Cell-DEVS model
+     * @tparam T the type used for representing time in a simulation.
+     * @tparam C the used for representing a cell ID.
+     * @tparam S the type used for representing a cell state.
+     * @tparam V the type used for representing a neighboring cell's vicinity. By default, it is set to integer.
+     * @tparam C_HASH the hash function used for creating unordered maps with cell IDs as keys.
+     *                By default, it uses the one defined in the standard library
+     */
+    template<typename T, typename C, typename S, typename V=int, typename C_HASH=std::hash<C>>
+    class cells_coupled: public cadmium::dynamic::modeling::coupled<T> {
+    public:
+
+        using CV = std::unordered_map<C, V, C_HASH>;
```

Comment:
whats CV? and CNV?

Reply:
Just aliases for reducing code size.
* CV is an unordered map {neighbor cell ID: vicinity} (just as above).
* CNV is a nested unordered map {target cell ID: {neighbor cell ID: vicinity}}

Vicinities are directed (i.e., A may be neighbor of B without B being neighbor of A).

Reply:
We don't pay per character we use, so its fine to spend a few more letters for the sake of clarity to those coming after, there is a big chance that someone else will contribute to this code in the future, the easier it is to read, the more probable someone is willing to extend i and do not go away before finishing. :)

Reply:
Again, I removed these aliases and substituted by a template class for defining unordered maps which key is a cell ID